app/assets/javascripts-es6/item-select.js

```diff
@@ -0,0 +1,37 @@
+$(document).ready(() => {
+  const items = document.getElementsByClassName("item");
+  for(let i=0; i<items.length; i++){
+    items[i].addEventListener("click", onItemSelect);
+  }
+});
+
+const onItemSelect = (e) => {
```

Comment:
You've moved this outside of the document.ready function scope. On one hand, that's good, because it can be done at load time rather than waiting until the document is ready. On the other hand, you may now be leaking this variable name into the global scope. It depends upon how your code is loaded.

Try accessing `window.onItemSelect` in your developer console. If it's defined, you should look into wrapping this code in an [IFFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression), or moving it into the above document.ready callback.

Reply:
Okay will read more on IFFE but this generally makes sense.